core/primitives-core/src/parameter.rs

```diff
@@ -0,0 +1,202 @@
+    ActionAddFullAccessKeyExecution,
+    ActionAddFunctionCallKeySendSir,
+    ActionAddFunctionCallKeySendNotSir,
+    ActionAddFunctionCallKeyExecution,
+    ActionAddFunctionCallKeyPerByteSendSir,
+    ActionAddFunctionCallKeyPerByteSendNotSir,
+    ActionAddFunctionCallKeyPerByteExecution,
+    ActionDeleteKeySendSir,
+    ActionDeleteKeySendNotSir,
+    ActionDeleteKeyExecution,
+
+    // Smart contract dynamic gas costs
+    WasmRegularOpCost,
+    WasmGrowMemCost,
+    WasmHostFunctionBase,
+    WasmContractLoadingBase,
+    WasmContractLoadingBytes,
+    WasmReadMemoryBase,
+    WasmReadMemoryByte,
+    WasmWriteMemoryBase,
+    WasmWriteMemoryByte,
+    WasmReadRegisterBase,
+    WasmReadRegisterByte,
+    WasmWriteRegisterBase,
+    WasmWriteRegisterByte,
+    WasmUtf8DecodingBase,
+    WasmUtf8DecodingByte,
+    WasmUtf16DecodingBase,
+    WasmUtf16DecodingByte,
+    WasmSha256Base,
+    WasmSha256Byte,
+    WasmKeccak256Base,
+    WasmKeccak256Byte,
+    WasmKeccak512Base,
+    WasmKeccak512Byte,
+    WasmRipemd160Base,
+    WasmRipemd160Block,
+    WasmEcrecoverBase,
+    WasmLogBase,
+    WasmLogByte,
+    WasmStorageWriteBase,
+    WasmStorageWriteKeyByte,
+    WasmStorageWriteValueByte,
+    WasmStorageWriteEvictedByte,
+    WasmStorageReadBase,
+    WasmStorageReadKeyByte,
+    WasmStorageReadValueByte,
+    WasmStorageRemoveBase,
+    WasmStorageRemoveKeyByte,
+    WasmStorageRemoveRetValueByte,
+    WasmStorageHasKeyBase,
+    WasmStorageHasKeyByte,
+    WasmStorageIterCreatePrefixBase,
+    WasmStorageIterCreatePrefixByte,
+    WasmStorageIterCreateRangeBase,
+    WasmStorageIterCreateFromByte,
+    WasmStorageIterCreateToByte,
+    WasmStorageIterNextBase,
+    WasmStorageIterNextKeyByte,
+    WasmStorageIterNextValueByte,
+    WasmTouchingTrieNode,
+    WasmReadCachedTrieNode,
+    WasmPromiseAndBase,
+    WasmPromiseAndPerPromise,
+    WasmPromiseReturn,
+    WasmValidatorStakeBase,
+    WasmValidatorTotalStakeBase,
+    WasmAltBn128G1MultiexpBase,
+    WasmAltBn128G1MultiexpElement,
+    WasmAltBn128PairingCheckBase,
+    WasmAltBn128PairingCheckElement,
+    WasmAltBn128G1SumBase,
+    WasmAltBn128G1SumElement,
+
+    // Smart contract limits
+    MaxGasBurnt,
+    MaxGasBurntView,
+    MaxStackHeight,
+    StackLimiterVersion,
+    InitialMemoryPages,
+    MaxMemoryPages,
+    RegistersMemoryLimit,
+    MaxRegisterSize,
+    MaxNumberRegisters,
+    MaxNumberLogs,
+    MaxTotalLogLength,
+    MaxTotalPrepaidGas,
+    MaxActionsPerReceipt,
+    MaxNumberBytesMethodNames,
+    MaxLengthMethodName,
+    MaxArgumentsLength,
+    MaxLengthReturnedData,
+    MaxContractSize,
+    MaxTransactionSize,
+    MaxLengthStorageKey,
+    MaxLengthStorageValue,
+    MaxPromisesPerFunctionCallAction,
+    MaxNumberInputDataDependencies,
+    MaxFunctionsNumberPerContract,
+    Wasmer2StackLimit,
+    MaxLocalsPerContract,
+}
+
+pub fn load_parameters_from_txt(arg: &[u8]) -> BTreeMap<Parameter, String> {
```

Comment:
```suggestion
pub(crate) fn load_parameters_from_txt(arg: &[u8]) -> BTreeMap<Parameter, String> {
```

I think it makes sense to avoid `pub` as much as possible. Cross-crate deps are the only deps that matter, so they should stand out.

Reply:
Hm, I defined this in `primitives-core` crate, where as all callers are in `primitives` crate. So your suggestion doesn't work out of the box. But is also seems wrong to have code define on crate A and only call it in crate B.

However, the type `Parameter` must be in `primitives-core` because it should be the new source-of-truth replacing some types in `core/primitives-core/src/profile.rs`. And `primitives-core` is not allowed to depend on other crates, according to our architecture.md.

So, in my latest commit, I moved all the logic to construct over to `primitives`. To still have the constructor syntax like `VMLimitConfig::from_parameters(params)`, I defined an extension trait `FromParameterTable`. (I can also rewrite to `params.vm_limit_config()` if you think that's better.)

This refactor sort of works but also brings new problems. I previously liked that each config struct had all constructors (for testing and for production) right next to each other, whereas now they are defined in different crates. And there are other things, like `wasmer2_stack_limit_default` that used to be private but now turned public in the refactor.

Not sure if it is an overall improvement or not.